nms/app/packages/magmalte/app/components/GatewayCommandFields.js

```diff
@@ -391,3 +407,133 @@ export function GenericCommandControls(props: ChildProps) {
     </div>
   );
 }
+
+type FileComponentProps = {
+  title: string,
+  content: string,
+  error?: string,
+};
+function FileComponent(props: FileComponentProps) {
+  const classes = useStyles();
+
+  // const content = props.content.split('\\n').map((v, i) => <><p key={i}>{v}</p></>);
+  const content = props.content.replace(/\\n/g, '\n');
+  return (
+    <List>
+      <ListItem> {props.title} </ListItem>
+      {props.error !== '""' && (
+        <ListItem>
+          <AltFormField label={''}>
+            <FormLabel data-testid="fileError" error>
+              {props.error}
+            </FormLabel>
+          </AltFormField>
+        </ListItem>
+      )}
+      <ListItem>
+        <textarea
+          data-testid="fileContent"
+          rows="8"
+          className={classes.jsonTextarea}
+          autoCapitalize="none"
+          autoComplete="none"
+          autoCorrect="none"
+          spellCheck={false}
+          value={content}
+        />
+      </ListItem>
+    </List>
+  );
+}
+
+const TROUBLESHOOTING_HINTS = {
+  FLUENTD_SUCCESS:
+    'Gateway contains fluentd parameters, Verify if \
+control proxy config contains the right fluentd address and port information, \
+typically fluend address is fluentd.<orc8r_domain_name> and fluend port is 24224',
+  FLUENTD_MISSING:
+    'Gateway is missing fluentd parameters, Add the \
+right fluentd address and port information to the control proxy config on the \
+gateway, typically fluend address is fluentd.<orc8r_domain_name> and fluend port\
+ is 24224',
+  AGG_API_SUCCESS:
+    'event and log aggregation API are returning successful \
+responses',
+};
+
+export function TroubleshootingControl(props: ChildProps) {
+  const {match} = useRouter();
+  const [
+    controlProxyContent,
+    setControlProxyContent,
+  ] = useState<generic_command_response>({});
+  const networkId = nullthrows(match.params.networkId);
+  const parameters = {
+    command: 'bash',
+    params: JSON.parse(`{
+      "shell_params" : [
+        "-c 'cat /etc/magma/control_proxy.yml'"
+      ]
+    }`),
```

Comment:
Just set the params as an object directly?

Reply:
Kind of silly question, flow complains here
```(property) parameters: {
    command: string;
    params: {
        shell_params: string[];
    };
}
{|
  command: string,
  params?: { [string]: { ... } }
|}
Cannot call `useMagmaAPI` with object literal bound to `params` because  array literal [1] is incompatible with  object type [2] in property `parameters.params.shell_params`.Flow(incompatible-call)```
Do you know how to circumvent it, flow seems to be incorrectly interpreting the contents of params as an array instead of an object type.